### DISCUSSION TITLE
Make fence-agents optional

### DISCUSSION THREAD

**Discussion by sbluhm** [cobbler/cobbler#3070] [Version specific questions] [answered]
Is it possible to make the fence-agents requirements optional?
I don't understand the actual usage or impact if fence-agents are not used (at least from my use cases).

Background:
Currently, RHEL9 does not ship with fence agents (and is still missing some requirements to build successfully)

**Comment by SchoolGuy** [comment, marked as answer]
So at the moment we use it for the whole power switching stack. If we package the functionality of this independently we can certainly make this optional.

The challenge for this is that on Cobbler Systems we have a `power_management_type` which is being validated through the power manager module, if we can somehow hide the power management attributes in case the `power_manager` module is unavailable then we have a way forward to make the `fence_agents` optional for Cobbler.
